sbpy/units/tests/test_core.py

```diff
@@ -96,3 +97,83 @@ def test_spectral_density_vega_bp(filename, fluxd, to, tol):
 def test_spectral_density_vega_synphot_import_fail():
     with mock.patch.dict(sys.modules, {'synphot': None}):
         assert spectral_density_vega(1 * u.um) == []
+
+
+@pytest.mark.parametrize('mag, wfb, f_sun, M_sun, ref', (
+    (3.4 * VEGAmag, None, None, None, 0.02865984),
+    (3.4 * VEGAmag, 'johnson_v', None, None, 0.02865984),
+    (3.4 * VEGAmag, 5500 * u.AA, None, None, 0.02774623),
+    (1.56644783e-09 * u.Unit('W/(m2 um)'), None,
+        1839.93273227 * u.Unit('W/(m2 um)'), None, 0.02865984),
+    (1.55728147e-24 * u.Unit('W/(m2 Hz)'), None,
+        1.86599755e-12 * u.Unit('W/(m2 Hz)'), None, 0.02809415),
+    (3.4 * VEGAmag, None, None, -26.77471503 * VEGAmag, 0.02865984),
+    (3.4 * u.ABmag, None, None, -26.77471503 * u.ABmag, 0.02865984),
+))
+def test_reflectance_ref(mag, wfb, f_sun, M_sun, ref):
+    """Test conversion from flux to reflectance
+
+    Use Ceres as the reference: H = 3.4 mag, radius = 460 km, average
+    bidirectional reflectance at zero phase angle 0.029 (~1/pi of geometric
+    albedo).
+    """
+    xsec = 6.648e5 * u.km**2
+    r = mag.to('1/sr', reflectance(cross_section=xsec, wfb=wfb, f_sun=f_sun,
+        M_sun=M_sun))
+    assert r.unit == u.sr**-1
+    assert np.isclose(r.value, ref)
+
+
+@pytest.mark.parametrize('mag, wfb, f_sun, M_sun, radius', (
+    (3.4 * VEGAmag, None, None, None, 460.01351274),
+    (3.4 * VEGAmag, 'johnson_v', None, None, 460.01351274),
+    (3.4 * VEGAmag, 5500 * u.AA, None, None, 452.62198065),
+    (1.56644783e-09 * u.Unit('W/(m2 um)'), None,
+        1839.93273227 * u.Unit('W/(m2 um)'), None, 460.01351274),
+    (1.55728147e-24 * u.Unit('W/(m2 Hz)'), None,
+        1.86599755e-12 * u.Unit('W/(m2 Hz)'), None, 455.45095634),
+    (3.4 * VEGAmag, None, None, -26.77471503 * VEGAmag, 460.01351274),
+    (3.4 * u.ABmag, None, None, -26.77471503 * u.ABmag, 460.01351274),
+))
+def test_reflectance_xsec(mag, wfb, f_sun, M_sun, radius):
+    """Test conversion from flux to reflectance
+
+    Use Ceres as the reference: H = 3.4 mag, radius = 460 km, average
+    bidirectional reflectance at zero phase angle 0.029 (~1/pi of geometric
+    albedo 0.09).
+    """
+    ref = 0.02865984 / u.sr
+    xs = mag.to('km2', reflectance(reflectance=ref, wfb=wfb, M_sun=M_sun))
+    ra = np.sqrt(xs/np.pi)
+    assert ra.unit == u.km
+    assert np.isclose(ra.value, radius)
+
+
+def test_reflectance_spec():
+    """Test conversion from flux spectrum to reflectance spectrum
+
+    Use Tempel 1 spectrum collected by Deep Impact HRI-IR at
+    2005-07-04 05:32 UT as the standard.  Data file
+    'data/hi05070405_9000036-avg-spec.txt'
```

Comment:
Add citation information here?

Reply:
Sure.  What's the best citation to this spectrum because you sent it to me :)

Reply:
It is in the file header.

Reply:
Yes, sorry for missing that.